designs/executables.md

```diff
@@ -0,0 +1,54 @@
+# Executable Extension Point
+
+The toolkit relies on external CLI's to provide common functionality - this pattern has been adopted
+to avoid duplicate work across multiple IDE integrations.
+
+The classic example is the SAM CLI which provides the majority of the toolkit's serverless functionality.
+
+## Common Features
+
+Each CLI/Executable that the toolkit integrates with will have several common features and thus it makes
+sense to create an extension-point abstraction to ease adding new CLIs in the future.
+
+Features that each CLI will require:
+* An ability to for the customer to configure the path/location of the executable
+  * Persistence of this setting
+* An ability for parts of the toolkit to be able to determine said path
+
+A CLI may also be able to 
+* Automatically resolve the path of the executable for the customer (e.g. find it locally, or resolve it from some remote location)
+* Determine if the executable is valid (e.g. can be executed, is a valid version etc)
+* Surface version information
+
+These optional features will be determined based on a set of mix-in interfaces.
+
+## Proposed Interfaces
```

Comment:
I would be interested in seeing some information on how these interfaces are used (for example, is there a corresponding manager that I would call to leverage these).